src/main/java/codeu/model/store/basic/UserStore.java

```diff
@@ -121,4 +128,101 @@ public boolean isUserRegistered(String username) {
   public void setUsers(List<User> users) {
     this.users = users;
   }
+
+  /**
+    * Returns the number of users currently recorded
+    */
+  public int getUserCount() {
+    return users.size();
+  }
+
+  /**
+    * Returns the users array
+    */
+  public List<User> getAllUsers(){
+    return users;
+  }
+
+  /**
+    * Returns the user with the most messages sent
+    * Pre : the the users list should already have the messageCount calculated
+    */
+  public String getTopUser(){
+    sortUserList();
+    return users.get(0).getName();
+  }
+
+  /**
+    * Iterate through all messages and get the count for the owner of ther messages
+    * Should only be called in PersistentDataStore when the messageCount attr does not exist
+    */
+  public int getMessageCountForUser(UUID uuid){
+    List<Message> messages = MessageStore.getInstance().getAllMessages();
+    int messageCount = 0;
+    for (Message message : messages) {
+      UUID id = message.getAuthorId();
+      if (id != null) {
+        if (uuid == id) {
+          messageCount++;
+        }
+      }
+    }
+    return messageCount;
+  }
+
+  /**
+    * Sort the users list by using sort in the library
+    */
+  public void sortUserList(){
+    Collections.sort(users, new Comparator<User>(){
+      public int compare(User a, User b){
+        return b.getMessageCount() - a.getMessageCount();
```

Comment:
I think this way is functionally correct, but 

`b.getMessageCount() > a.getMessageCount`

would both be a better representation of what you're doing and also more efficient.  

Or even better, you can use b.getMessagesCount().compareTo(a.getMessagesCount()) and level the language's builtin ability to compare data types.

Reply:
I see, but I don't understand why > would be more efficient

Reply:
I'm confused with the compareTo method, i thought that method is specifically for String comparisons only